testing/trino-tests/src/test/java/io/trino/execution/TestTableRedirection.java

```diff
@@ -363,6 +366,66 @@ public void testShowColumns()
                 .hasMessageContaining("Table redirections form a loop");
     }
 
+    @Test
+    public void testInsert()
+    {
+        assertUpdate(
+                getSession(),
+                format("INSERT INTO %s.%s VALUES (5, 6)", SCHEMA_ONE, VALID_REDIRECTION_SRC),
+                1,
+                // Verify the insert plan instead of through a successive SELECT, because insertion is a no-op for Mock connector
```

Comment:
Can we also have a real end-to-end test as well?

Reply:
We can add one with the #8340, will that be okay?

Reply:
Absolutely.